cmd/app/main.go

```diff
@@ -2,78 +2,162 @@ package main
 
 import (
 	"context"
-	"fmt"
+	"errors"
 	"log"
 	"net/url"
 	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
 	"parabellum.crawler/internal/crawler"
-)
-
-//TODO: remove const block after final implementation
-const (
-	mockSiteName = "http://httpstat.us/"
-	mockFileName = "results.log"
-)
+	"parabellum.crawler/internal/pubsub"
+	"parabellum.crawler/model"
 
-const (
-	DEFAULT_TIMEOUT = time.Minute
-	NUM_OF_THREADS  = 50
-	MAX_DEPTH       = 3
+	"github.com/joho/godotenv"
 )
 
 type Config struct {
-	Crawler *crawler.Crawler
+	Crawler   *crawler.Crawler
+	Consumer  *pubsub.Consumer
+	Producers map[string]*pubsub.Producer
+}
+
+func init() {
+	err := godotenv.Load(pathToEnvFile)
+	if err != nil {
+		log.Panicln("Error loading .env file: ", err)
+	}
 }
 
 func main() {
-	//receive input - site to crawl
-	//TODO: unmock site name receiving, when get input data format
-	providedURL, err := url.Parse(mockSiteName)
+	app := new(Config)
```

Comment:
You can use chain here, like : `app := new(Config).initPubSub()`

Reply:
This comment was duplicated somehow. Replied above